### PR TITLE
Sitemap e2e test

### PR DESCRIPTION
**Pull request by boodland**
### Issue link / number:
#1089

### What changes did you make?
Fixed sitemap urls
Created sitemap e2e test

### Why did you make the changes?
So the page is covered with test and help us to validate it is working when making changes or migrating it to app router

### Did you run tests?
Yes

### CODE REVIEW

File: cypress/integration/tests/sitemap.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+describe('Sitemap should have', () => {
+  const SITEMAP_URL = '/sitemap.xml';
+  const BLOOM_URL = 'https://bloom.chayn.co';
+  before(() => {
+    cy.cleanUpTestState();
+  });
+
+  it('bloom url', () => {
+    cy.request(SITEMAP_URL).its('body').should('include', BLOOM_URL);
```
Comment (eleanorreem) [inline cypress/integration/tests/sitemap.cy.tsx:9]: @boodland I think this test is too detailed. I reckon you just need to check it exists with one or two tests on urls that won't change.

File: cypress/integration/tests/sitemap.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  });
+  it('about our courses url', () => {
+    cy.request(SITEMAP_URL)
```
Comment (eleanorreem) [inline cypress/integration/tests/sitemap.cy.tsx:12]: This test should be removed

File: pages/sitemap.xml.tsx [hunk 3]
```diff
@@ -10,10 +10,10 @@ async function generateSiteMap({ locales }: any) {
             <loc>https://bloom.chayn.co</loc>
         </url>
         <url>
-            <loc>https://bloom.chayn.co/therapy/about-our-courses</loc>
+            <loc>https://bloom.chayn.co/about-our-courses</loc>
```
Comment (eleanorreem) [inline pages/sitemap.xml.tsx:13]: remove this from the sitemap

File: pages/sitemap.xml.tsx [hunk 4]
```diff
@@ -10,10 +10,10 @@ async function generateSiteMap({ locales }: any) {
...
         </url>
         <url>
-            <loc>https://bloom.chayn.co/therapy/meet-the-team</loc>
+            <loc>https://bloom.chayn.co/meet-the-team</loc>
```
Comment (eleanorreem) [inline pages/sitemap.xml.tsx:16]: thanks for this correction

File: pages/sitemap.xml.tsx [hunk 5]
```diff
@@ -22,10 +22,11 @@ async function generateSiteMap({ locales }: any) {
             <loc>https://bloom.chayn.co/therapy/confirmed-session</loc>
         </url>
         ${courses
-          .map((course: any) => {
+          .map((course: { locale: string; params: { slug: string; sessionSlug: string } }) => {
+            const locale = course.locale !== 'en' ? `/${course.locale}` : '';
             return `
                     <url>
-                        <loc>${`https://bloom.chayn.co/courses/${course.params.slug}/${course.params.sessionSlug}`}</loc>
+                        <loc>${`https://bloom.chayn.co${locale}/courses/${course.params.slug}/${course.params.sessionSlug}`}</loc>
```
Comment (eleanorreem) [inline pages/sitemap.xml.tsx:29]: thanks for this

**Comment by eleanorreem** [comment]
Thanks @boodland ! 1 small change!

File: cypress/integration/tests/sitemap.cy.tsx (hunk 1 above)
Comment (boodland) [reply cypress/integration/tests/sitemap.cy.tsx:9]: Test cases simplified

**Comment by boodland** [comment]
@eleanorreem changes applied!

**Review by eleanorreem — APPROVED** [review]

**Pull request merged**